### DISCUSSION TITLE
Error with 'flet build apk' and sqlite

### DISCUSSION THREAD

**Discussion by marcrr79907** [flet-dev/flet#3817] [Q&A] [answered]
### Question

Please, I need to create the .apk file of a project that receives data from an SQLite database, but when executing the "flet build apk" command, an error occurs.

### Error message

```bash
[14:55:01] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.23.2 ✅             
[14:55:02] Customized app icons and splash images ✅                                                             
[14:55:11] Generated app icons ✅                                                                                
[14:55:16] Generated splash screens ✅                                                                           
[14:55:24] Flet app package app/app.zip was not created.
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Can you use `--verbose` when running the build command?

**Reply by marcrr79907** [reply]
This is the result with 'flet build apk -v' 
flet build apk -v
[06:59:40] Flutter bootstrap directory: C:\Users\Marc\AppData\Local\Temp\flet_flutter_build_xm12WPtgjd                 
           Additional Flutter dependencies:                                                                            
           {}                                                                                                          
[07:00:19] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.23.2 ✅                   
           Copying D:\Programation\PyProjects\TarifarioCopextel\assets\icon.png to                                     
           C:\Users\Marc\AppData\Local\Temp\flet_flutter_build_xm12WPtgjd\images                                       
           Customized app icons and splash images ✅                                                                   
           Run subprocess: ['D:\\Programation\\Resources\\flutter_windows_3.22.2-stable\\flutter\\bin\\dart.BAT',      
           'run', 'flutter_launcher_icons']                                                                            
(  ●   ) Generating app icons ⏳...  Resolving dependencies in `C:\Users\Marc\AppData\Local\Temp\flet_flutter_build_xm12(    ● ) Generating app icons(20.5s)
(    ● ) Generating app icons ⏳... 
(   ●  ) Generating app icons ⏳... Got dependencies in `C:\Users\Marc\AppData\Local\Temp\flet_flutter_build_xm12WPtgjdd`.
(   ●  ) Generating app icons (7.6s)
Built flutter_launcher_icons:flutter_launcher_icons.
(   ●  ) Generating app icons ⏳...   ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════

• Creating default icons Android
( ●    ) Generating app icons ⏳...  • Overwriting the default Android launcher icon with a new icon
(  ●   ) Generating app icons ⏳...  • Overwriting default iOS launcher icon with new icon
( ●    ) Generating app icons ⏳...  Creating Icons for Web...
(   ●  ) Generating app icons ⏳...  Creating Icons for Windows...
(    ● ) Generating app icons ⏳...  Creating Icons for MacOS...
( ●    ) Generating app icons ⏳... 
✓ Successfully generated launcher icons
[07:00:51] Generated app icons ✅                                                                                      
           Run subprocess: ['D:\\Programation\\Resources\\flutter_windows_3.22.2-stable\\flutter\\bin\\dart.BAT',      
           'run', 'flutter_native_splash:create']                                                                      
(    ● ) Generating splash screens ⏳... 
Built flutter_native_splash:create.
( ●    ) Generating splash screens ⏳...  [Android] Creating default splash images
(●     ) Generating splash screens ⏳...  [Android] Creating dark mode splash images
(    ● ) Generating splash screens ⏳... [Android] Creating default android12splash images
(    ● ) Generating splash screens ⏳...  [Android] Creating dark mode android12splash images
(     ●) Generating splash screens ⏳...  [Android] Updating launch background(s) with splash image path...
[Android]  - android/app/src/main/res/drawable/launch_background.xml
( ●    ) Generating splash screens ⏳... [Android]  - android/app/src/main/res/drawable-night/launch_background.xml    
[Android]  - android/app/src/main/res/drawable-v21/launch_background.xml
[Android]  - android/app/src/main/res/drawable-night-v21/launch_background.xml
[Android] Updating styles...
[Android]  - android/app/src/main/res/values-v31/styles.xml
[Android] No android/app/src/main/res/values-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-v31/styles.xml and adding it to your Android project
[Android]  - android/app/src/main/res/values-night-v31/styles.xml
[Android] No android/app/src/main/res/values-night-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-night-v31/styles.xml and adding it to your Android project
[Android]  - android/app/src/main/res/values/styles.xml
[Android]  - android/app/src/main/res/values-night/styles.xml
[iOS] Creating  images
( ●    ) Generating splash screens ⏳...  [iOS] Creating dark mode  images
(   ●  ) Generating splash screens ⏳...  [iOS] Updating ios/Runner/Info.plist for status bar hidden/visible
(    ● ) Generating splash screens ⏳...  [Web] Creating images
(●     ) Generating splash screens ⏳...  [Web] Creating images
(     ●) Generating splash screens ⏳...  [Web] Creating background images
[Web] Creating CSS
(    ● ) Generating splash screens ⏳...  [Web] Updating index.html
rc\AppData\Local\Temp\serious_python_tempa86e5df2
(   ●  ) Packaging Python app ⏳... Error: FileSystemException: Failed to decode data using encoding 'utf-8', path = 'C:\Users\Marc\AppData\Local\Temp\serious_python_tempa86e5df2\requirements.txt'
Deleting temp directory C:\Users\Marc\AppData\Local\Temp\serious_python_tempa86e5df2
[07:01:08] Deleting Flutter bootstrap directory C:\Users\Marc\AppData\Local\Temp\flet_flutter_build_xm12WPtgjd                                                
           Flet app package app/app.zip was not created.

**Reply by ndonkoHenri** [reply, marked as answer]
Next time when sharing logs, please use `--verbose` or `--v` for maximum verbosity.
The issue can be read from your logs:
> Error: FileSystemException: Failed to decode data using encoding 'utf-8', path = 'C:\Users\Marc\AppData\Local\Temp\serious_python_tempa86e5df2\requirements.txt'

So, you simply have to make sure your requirements file is properly encoded (in utf8).
You can search online ([example](https://support.meistertask.com/hc/en-us/articles/4406395262354-How-Do-I-Encode-My-CSV-File-Using-the-UTF-8-Format)) on how to do that. Worst case, you might have to delete the current file, and create a new one, encoded in the proper format.

**Reply by marcrr79907** [reply]
Thank you very much bro, that worked
